server/src/main/java/org/elasticsearch/cluster/metadata/AliasOrIndex.java

```diff
@@ -154,12 +154,18 @@ void addIndex(IndexMetaData indexMetaData) {
         }
 
         public void computeAndValidateWriteIndex() {
-            List<IndexMetaData> writeIndices = referenceIndexMetaDatas.stream()
-                .filter(idxMeta -> Boolean.TRUE.equals(idxMeta.getAliases().get(aliasName).writeIndex()))
-                .collect(Collectors.toList());
-            if (referenceIndexMetaDatas.size() == 1) {
-                writeIndex.set(referenceIndexMetaDatas.get(0));
-            } else if (writeIndices.size() == 1) {
+            final List<IndexMetaData> writeIndices;
+            if (referenceIndexMetaDatas.size() > 1) {
+                writeIndices = referenceIndexMetaDatas.stream()
+                    .filter(idxMeta -> Boolean.TRUE.equals(idxMeta.getAliases().get(aliasName).writeIndex()))
+                    .collect(Collectors.toList());
+            } else if(Boolean.FALSE.equals(referenceIndexMetaDatas.get(0).getAliases().get(aliasName).writeIndex()) == false) {
```

Comment:
Up to you but I find it more intuitive to collect all the explicit write indices (i.e. flag is true) regardless of size if that is empty we check if `referenceIndexMetaDatas` is of size 1 and if the first value has ` null` for the flag, if so we set `writeIndices` to referenceIndexMetaDatas and be happy.